src/components/SliderContainer.ts

```diff
@@ -0,0 +1,198 @@
+        });
+    }
+
+    componentWillReceiveProps(newProps: SliderContainerProps) {
+        this.resetSubscriptions(newProps.mxObject);
+        this.updateValues(newProps.mxObject);
+    }
+
+    componentWillUnmount() {
+        this.unSubscribe();
+    }
+
+    private getAttributeValue(
+        contextObject: mendix.lib.MxObject,
+        attributeName: string,
+        defaultValue?: number): number | undefined {
+        if (contextObject && attributeName) {
+            if (contextObject.get(attributeName)) {
+                return parseFloat(contextObject.get(attributeName) as string);
+            }
+        }
+
+        return defaultValue;
+    }
+
+    private updateValues(contextObject: mendix.lib.MxObject) {
+        this.setState({
+            maximumValue: this.getAttributeValue(contextObject, this.props.maxAttribute),
+            minimumValue: this.getAttributeValue(contextObject, this.props.minAttribute),
+            stepValue: this.getAttributeValue(contextObject, this.props.stepAttribute, this.props.stepValue),
+            value: this.getAttributeValue(contextObject, this.props.valueAttribute)
+        });
+    }
+
+    private onUpdate(value: number) {
+        const { mxObject, valueAttribute } = this.props;
+        const { maximumValue } = this.state;
+        if (value || value === 0) {
+            if ((maximumValue || maximumValue === 0) && (value > maximumValue)) {
+                mxObject.set(valueAttribute, maximumValue);
+            } else {
+                mxObject.set(valueAttribute, value);
+            }
+        }
+    }
+
+    private handleAction(value: number) {
+        if (value || value === 0) {
+            this.executeMicroflow(this.props.onChangeMicroflow, this.props.mxObject.getGuid());
+        }
+    }
+
+    private executeMicroflow(actionname: string, guid: string) {
+        if (actionname) {
+            window.mx.ui.action(actionname, {
+                error: (error) => window.mx.ui.error(
+                    `An error occurred while executing microflow: ${actionname}: ${error.message}`
+                ),
+                params: {
+                    applyto: "selection",
+                    guids: [ guid ]
+                }
+            });
+        }
+    }
+
+    private resetSubscriptions(contextObject: mendix.lib.MxObject) {
+        this.unSubscribe();
+
+        if (contextObject) {
+            this.subscriptionHandles.push(window.mx.data.subscribe({
+                callback: () => this.updateValues(contextObject),
+                guid: contextObject.getGuid()
+            }));
+            [
+                this.props.valueAttribute,
+                this.props.maxAttribute,
+                this.props.minAttribute,
+                this.props.stepAttribute
+            ].forEach((attr) =>
+                this.subscriptionHandles.push(window.mx.data.subscribe({
+                    attr,
+                    callback: () => this.updateValues(contextObject),
```

Comment:
You're making new lamdas for each `attr` while they actually are all identical. Extract and reuse.